.github/scripts/test_open_data_assets.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env python3
+# Run tests on our open data assets to confirm that their row counts match
+# the row counts of the underlying Athena tables when grouped by year.
+#
+# Requires one positional argument, the path to a dbt manifest that can be
+# used to parse exposure data representing open data assets and their
+# dependency graph.
+#
+# Example:
+#
+#   cd dbt
+#   python3 ../.github/scripts/test_open_data_assets.py target/manifest.json
+
+import contextlib
+import datetime
+import io
+import json
+import sys
+import typing
+
+import agate
+import requests
+from dbt.cli.main import dbtRunner
+
+# API URL and params that we can use to grab counts by year for each asset
+ASSET_API_URL = "https://datacatalog.cookcountyil.gov/resource/{asset_id}.json"
+ASSET_API_QUERY_PARAMS = {"$query": "SELECT COUNT(*),year GROUP BY year"}
+DBT = dbtRunner()
+
+# Data for the most recent two years are permitted to be slightly different,
+# according to this buffer value
+BUFFER = 0.02
+
+
+def main() -> None:
+    """Entrypoint for the script."""
+    dbt_manifest_path = sys.argv[1]
+    with open(dbt_manifest_path) as dbt_manifest_file:
+        dbt_manifest_json = json.loads(dbt_manifest_file.read())
+
+    diffs: typing.List[typing.List[typing.Dict]] = []
+
+    for exposure in dbt_manifest_json["exposures"].values():
+        asset_name = exposure["label"]
+        asset_url = exposure["url"]
+        asset_id = asset_url.split("/")[-1]
+
+        # For now, assume just one dependency table, since all of our open
+        # data assets currently have a 1:1 relationship with a view.
+        fq_source_model_name = exposure["depends_on"]["nodes"][0]
+
+        # Dependency tables in the DAG are fully qualified, so we have to
+        # strip the prefix to get the schema and tablename.
+        source_model_name = ".".join(fq_source_model_name.split(".")[2:])
+
+        print(f"Comparing asset '{asset_name}' to model '{source_model_name}'")
+
+        asset_api_url = ASSET_API_URL.format(asset_id=asset_id)
+        response = requests.get(asset_api_url, params=ASSET_API_QUERY_PARAMS)
+        if not response.ok:
+            print("Encountered error in request to asset endpoint")
+            response.raise_for_status()
+
+        asset_row_counts_by_year = response.json()
+
+        dbt_output = io.StringIO()
+        with contextlib.redirect_stdout(dbt_output):
```

Comment:
This is not a super clean way of returning data from the underlying query, but as far as I can tell, there is not yet a way of returning rows to a macro invoked via `dbtRunner.invoke()`, so we have to rely on printing to stdout and then capturing that output.